doc/userDocs/usersGuide.md

```diff
@@ -259,6 +263,118 @@ but it will not stress the parts of the network stack that manage directing traf
+
+
+##### Customizing the Generation of Namespace Names
+
+Weathervane generates the namespace names using three components:
+* A namespace prefix, which defaults to **auction**.
+* A unique identifier, which is __w1__ for the drivers and __w1i*n*__ for each application 
+  instance __*n*__.
+* A namespace suffix, which is blank by default.
+
+It is possible to change the strings used for the prefix and suffix using the parameters 
+`namespacePrefix` and `namespaceSuffix`.  These parameters are placed in the configuration 
+file within the definition of the Kubernetes cluster.
+
+For example, the following configuration file excerpt will use namespace names of the 
+form __appw1i*n*small__ for the cluster named *appCluster* and __driverw1small__ for 
+the cluster named *driverCluster*.
+
+```json
+"kubernetesClusters" : [ 
+  { 
+    "name" : "appCluster", 
+    "kubeconfigFile" : "/root/.kube/config",
+    "kubeconfigContext" : "cluster-context-1",
+    "namespacePrefix" : "app",
+    "namespaceSuffix" : "small",
+  },
+  { 
+    "name" : "driverCluster", 
+    "kubeconfigFile" : "/root/.kube/config",
+    "kubeconfigContext" : "cluster-context-2",
+    "namespacePrefix" : "driver",
+    "namespaceSuffix" : "small",
+  },
+],
+```
+
+The ability to specify a unique suffix is particularly useful when you are doing tests with different 
+configuration sizes on the same Kubernetes cluster.  Using a unique `namespaceSuffix` for 
+each configuration size will prevent the data services from being reloaded each time 
+you switch configuration size.
+
+##### Specifying Namespace Names
+
+There are situations in which namespace names must be used that do not fit the default or customized 
+name generation process.  Some examples are:
+* The namespace names must adhere to specific conventions
+* The namespaces are created and named by an administrator
+* The namespaces are created and named by an automated provisioning system
+
+In these cases, it is possible to use the `namespaces` parameter to provide a list of namespace names for each 
+Kubernetes cluster.  Weathervane will use namespace names from that list rather than 
+generating the namespace names.
+
+The following configuration file excerpt shows how to specify a list of namespace names. In 
+this example Weathervane will use *devNs1* and *devNs2* as the names for the workload driver 
+and application namespaces. 
+
+```json
+"kubernetesClusters" : [ 
+  { 
+    "name" : "k8sCluster", 
+    "kubeconfigFile" : "/root/.kube/config",
+    "kubeconfigContext" : "cluster-context-1",
+    "namespaces" : ["devNs1", "devNs2",],
+  },
+],
+```
+When specifying namespace names, you must provide at least one namespace name for each 
+application instance you plan to run on the cluster, plus one namespace name for the 
+workload drivers.  This means that if you are running the drivers and applications 
+on the same cluster you will need to specify a minimum of two namespaces. In this case 
```

Comment:
"In this case the first namespace will be used for the workload drivers and others for the application instances"
Is the first namespace in the list always used for the driver and the second for application instances?, when one cluster is used for both? Does it alternate between driver and application, such as driver1, app1, driver2, app2 and so forth?

Reply:
If the drivers and the apps are running on the same cluster, and use the same kubernetesCluster, then the first will always be used for the workload driver, and the others for the application instances.  Because we are not documenting the ability to have two workloads, there will never be more than one namespace used for workload driver.  If the driver and apps are running on different clusters, then there will be separate lists for each cluster.

Note that it is possible to specify two kubernetesCluster objects for the same cluster, with different names, and then give each a separate list of namespaces.  So you could have one cluster definition with a namespace for the driver, and another with namespaces for the applications, both pointing to the same kubeconfig and context.  I just didn't want to get into that much detail until we see that someone needs the functionality.  

There is no guarantee that the first namespace will be used for the drivers in all future versions though, so I think that it will be safest to take this line out.